core/hash_map.h

```diff
@@ -74,13 +78,14 @@ class HashMapHahserDefault {
  * @param TKey  Key, search is based on it, needs to be hasheable. It is unique in this container.
  * @param TData Data, data associated with the key
  * @param Hasher Hasher object, needs to provide a valid static hash function for TKey
+ * @param Comparator comparator object, needs to be able to safely compare two TKey values
```

Comment:
And no, the comment I was referring is this

```
   101	// NOTE: Because NaN != NaN, a map can contain any
   102	// number of (mostly useless) entries keyed with NaNs.
   103	// To avoid long hash chains, we assign a random number
   104	// as the hash value for a NaN.
```

(which is also the link I gave points to)